src/python/bot/fuzzers/syzkaller/engine.py

```diff
@@ -93,6 +98,35 @@ def _create_temp_corpus_dir(self, name):
     engine_common.recreate_directory(new_corpus_directory)
     return new_corpus_directory
 
+  def save_corpus(self, source, destination):
+    """Saves syzkaller to folder so it is backed up to the cloud.
+
+    Args:
+      source: Folder where syzkaller corpus is.
+      destination: Folder where the corpus is synced with the cloud.
+    """
+    source_file = os.path.join(source, CORPUS_FILENAME)
+    shell.create_directory(destination)
+    target_file = os.path.join(destination,
+                               settings.get_device_codename() + '.db')
+    if os.path.isfile(source_file):
+      shutil.copy(source_file, target_file)
+
+  def init_corpus(self, source, destination):
+    """Uses corpus from the ckoud to initialize syzkaller corpus.
+
+    Args:
+      source: Folder where the corpus is downloaded from the cloud.
+      destination: Folder where syzkaller will be looking for corpus.
+    """
+    source_file = os.path.join(source, settings.get_device_codename() + '.db')
+    shell.create_directory(destination)
+    destination_file = os.path.join(destination, CORPUS_FILENAME)
+    if os.path.isfile(source_file) and (
+        not os.path.exists(destination_file) or
+        (os.path.getsize(source_file) > os.path.getsize(destination_file))):
```

Comment:
Maybe just use "!=" rather than > ? Does the corpus ever get smaller (e.g. minimized) during fuzzing ?

Reply:
Yes there is a smaller corpus case when device is in bad state, like console cable not working. Max spector gave that.